[WIP] BLDR setup: Remove hardcoded github credentials

indradhanush opened the pull request:
This commit adds an option to create a new file under
`$PROJECT_ROOT/src/.secrets/habitat-env` that can store custom github
keys and secrets. If this file is not present it defaults to the
current behaviour of using the hardcoded keys.

The changes affect the Vagrant setup and this also includes
instructions for setting up BLDR under Vagrant.

indradhanush committed:
BLDR setup: Remove hardcoded github credentials

This commit adds an option to create a new file under
$PROJECT_ROOT/src/.secrets/habitat-env that can store custom github
keys and secrets. If this file is not present it defaults to the
current behaviour of using the hardcoded keys.

The changes affect the Vagrant setup and this also includes
instructions for setting up BLDR under Vagrant.

robertgzr committed:
Vagrantfile: add web ui & k8s ports

schu commented:
The test instructions (work in progress) can be found here: https://github.com/kinvolk/habitat/blob/92bb1282ef6375d03fa722f389b12d6ae216ec8a/scripts/vagrant/README.md

robertgzr commented on scripts/vagrant/README.md:
````diff
@@ -0,0 +1,68 @@
+export GITHUB_WEBHOOK_SECRET=""
+```
+
+- Ask a cowrecker for secrets (easy). Or generate your own (painful).
````
s/crowrecker/coworker 😆

indradhanush replied:
That's intentional. Because we "wreck" things together. 😛

schu commented:
@indradhanush can you add the forwardings Robert posted to your Vagrant file so that your pull request is testable?

robertgzr commented on scripts/vagrant/README.md:
```diff
@@ -0,0 +1,68 @@
+
+- [Create a Github app](https://github.com/settings/apps) and download the private key from the app's page.
+- Put it under `$PROJECT_ROOT/.secrets/builder-github-app.pem`
+- Create a new file at `$PROJECT_ROOT/.secrets/habitat-env` and write the following into it:
```
Maybe call it `builder-env` since the variables mainly affect builder and not the rest of habitat, no?

Also.. could https://github.com/kinvolk/habitat/tree/master/support/bldr.env be used for these?

schu replied:
We need to use something which is not in git, as it is user specific configuration.

Details like file name / location can be discussed later, let's focus on getting it to work reliable.

schu commented:
> For the demo we want to run bldr, the web UI and a kube-spawn cluster in the same Vagrantfile.

Some discussion on that happended here: https://github.com/kinvolk/habitat-builder-to-kubernetes/issues/6 We talked a bit more OOB and said spawning a second, Fedora 27 VM, from the Habitat Vagrant file might is better than changing the Habitat dev VM to 17.10. To keep changes to upstream and ours more separate from the beginning.

Commit:
Procfile: let the web interface run again

Since 0.39.0, support/Procfile doesn't have an entry to run the web
interface, so `sudo -E make bldr-run` does not run `lite-server` at all.
Let's bring the line back to Procfile for easy testing.

Commit:
README: add a note about running builder_web.sh

indradhanush committed:
README: Link to secret gist containing keys

Commit:
rxjs/Subscription

Commit:
README: describe more about setting up app ID, client ID

schu committed:
Update README

schu committed:
Update README

Commit:
README: set env variables for github auth token, origin

set env variables as well as necessary permissions on the github page.

schu committed:
Update README

schu committed:
README: mention public key workaround

schu committed:
Update README

Commit:
README: describe about timer issue, github auth issue

Commit:
README: create a directory /home/krangschnak/.hab/cache/keys

schu commented:
We moved the code / discussion upstream: https://github.com/habitat-sh/habitat/pull/4161
